lib/explorer/data_frame.ex

```diff
@@ -2078,6 +2086,24 @@ defmodule Explorer.DataFrame do
                 "but instead it returned a LazySeries of type " <>
                 inspect(dtype)
 
+      [%Series{dtype: :boolean, data: %LazySeries{}} | _rest] = lazy_series ->
+        first_non_boolean =
+          Enum.find(lazy_series, &(!match?(%Series{dtype: :boolean, data: %LazySeries{}}, &1)))
+
+        if is_nil(first_non_boolean) do
+          series =
+            Enum.reduce(lazy_series, fn x, acc ->
+              Explorer.Backend.LazySeries.binary_and(acc, x)
+            end)
+
+          Shared.apply_impl(df, :filter_with, [df, series.data])
+        else
+          raise ArgumentError,
+                "expecting the function to return a single or a list of boolean LazySeries, " <>
+                  "but instead it contains:\n#{inspect(first_non_boolean)}\n" <>
+                  "that is of dtype #{inspect(first_non_boolean.dtype)}"
```

Comment:
I would remove the rest because if may not be a series at all?

```suggestion
                  "but instead it contains:\n#{inspect(first_non_boolean)}"
```